models/enterprise_sonic/snmp/deleted_example_01.txt

```diff
@@ -1,28 +1,31 @@
 # Using deleted
-
+#
 # Before state:
-# -------------
+#---------------
 #
-#rno-dctor-1ar01c01sw02# show snmp-server community
+# show running-configuration | grep snmp
 #
-#         Community Name                     Group Name
-#-------------------------------- --------------------------------
-#private                          ro
-#public                           ro
+# snmp-server user user1 group group-lab encrypted auth
+# md5 auth-password
+# U2FsdGVkX18J+L+L9OyQYWpAkGUrTgcg/6xzZoDjCbQw1ISHJ5mxmxrYZgQypEUXDeNe6rBupsc9sVDJBKxrwA==
+# priv aes-128 priv-password U2FsdGVkX1/Xs+ffZvdV9YzfyGHgIJ+zkLRPfF3/WgYIE1S4Ribvbzhu5chpHHI7ooCBpcVxYZotAXDzgetxvQ==
 #
-- name: Delete a single SNMP community
+
+- name: Delete specific option from snmp-server configuration
   sonic_snmp:
-    config:
-      - index: "private"
-        securityName: "ro"
-    state: delete
+      config:
+          user:
+              - name: user1
+                auth: 
+                  auth_type: md5
+                  key: U2FsdGVkX18J+L+L9OyQYWpAkGUrTgcg/6xzZoDjCbQw1ISHJ5mxmxrYZgQypEUXDeNe6rBupsc9sVDJBKxrwA==
+                priv:
+                  aes_key: aes-128
+                encrypted: True
```

Comment:
Because "encrypted" is set to True, the unencrypted value of the aes_key would be rejected by SONiC. To show this example correctly, either an "encrypted" value of "false" (not capitalized) would need to be specified along with an unencrypted md5 key or an encrypted aes_key would need to be specified.

But, more importantly, SONiC doesn't really support deletion of individual attributes for a given "user". (It explicitly rejects an attempt to do this.)

I would recommend changing this example from "Delete specific option ..." (which is not supported) to "Delete a sub-section ..." by specifying only the "name" value and none of the other options here. That would cause deletion of configuration for the user named "user1".

It may be useful to show for this example that if there are multiple users (e.g. "user1" and "user2") configured, it is possible to delete just "user1" and leave "user2" configured. Then for the second "deleted" example, deletion of the entire "user" section of the configuration could be demonstrated.